subscription.go

```diff
@@ -140,6 +154,49 @@ func (r *marathonClient) RegisterSubscription() error {
 	return nil
 }
 
+func (r *marathonClient) registerSSESubscription() error {
+	// Prevent multiple SSE subscriptions
+	if r.subscribedToSSE {
+		return nil
+	}
```

Comment:
@eicca Good point. But I see another issue here: `registerSSESubscription()` is only called from within `RegisterSubscription()` which in turn is *only* called from `AddEventsListener()` which is already synchronised. The issue is that `RegisterSubscription` itself is being exported while it is not defined in `Marathon` interface. I think that we could just rename `RegisterSubscription` to `registerSubscription` and that's it.